tools/c7n_azure/c7n_azure/actions/notify.py

```diff
@@ -89,5 +90,17 @@ def send_data_message(self, message, session):
             return self.send_to_azure_queue(queue_uri, message, session)
 
     def send_to_azure_queue(self, queue_uri, message, session):
-        queue_service, queue_name = StorageUtilities.get_queue_client_by_uri(queue_uri, session)
-        return StorageUtilities.put_queue_message(queue_service, queue_name, self.pack(message)).id
+        try:
+            queue_service, queue_name = StorageUtilities.get_queue_client_by_uri(queue_uri, session)
+            return StorageUtilities.put_queue_message(
+                queue_service,
+                queue_name,
+                self.pack(message)).id
+        except AzureHttpError as e:
+            if e.status_code == 403:
+                self.log.error("Access error while putting message to the queue. "
```

Comment:
Alternative message: `Access Error: Storage Queue Data Contributor Role is required to enqueue messages.`